Desktop_Interface/functiongencontrol.h

```diff
@@ -15,10 +15,27 @@ class functionGenControl : public QLabel
 {
     Q_OBJECT
 public:
+
+	struct ChannelData
+	{
+		unsigned char* samples = NULL;
```

Comment:
A lot of leaks and memcpy hack in the USB driver could just magically disappear if this was a std::vector.

Performance is not too critical in this instance as the length of samples is under 512 bytes (that's how much RAM I could spare on the microcontroller for function generator data; the two 512-byte buffers take up 1/4 of the XMEGA's 4kB SRAM!), and each change is followed by a blocking call to a libusb API anyway, which transfers these bytes over a 12Mb/s USB link (memory copies on a fairly standard CPU have bandwidth in the single-digit GB/s).

Reply:
That's a great point! I don't know how it didn't even cross my mind to use a vector instead.

I got a part time job doing php development recently maybe that's clouding my judgement hahaha